server/tests/src/test/java/org/infinispan/server/security/authorization/AbstractAuthorization.java

```diff
@@ -117,6 +117,8 @@ public void testRestAdminCanDoEverything() {
       RestCacheClient adminCache = getServerTest().rest().withClientConfiguration(restBuilders.get(TestUser.ADMIN)).withCacheMode(CacheMode.DIST_SYNC).create().cache(getServerTest().getMethodName());
       sync(adminCache.put("k", "v"));
       assertEquals("v", sync(adminCache.get("k")).getBody());
+
+      assertStatus(OK, adminCache.distribution());
```

Comment:
I think it is missing a test to check the non admin are unable to access the `distribution`

Reply:
I've follow the same approach as the `stats` endpoint, so all type have access. I'll update the test.

Reply:
@jabolina so, you are changing to `monitor`?

Reply:
Currently the cache stats endpoint is accessible by all types, should this one for distribution and stats be only monitor? I can change the distribution endpoint in this PR and the stats in a follow-up.

Reply:
@jabolina do you mean `org.infinispan.rest.resources.CacheResourceV2#getCacheStats()`? Or is it other cache stats?
It is not accessible by all types, it requires the `MONITOR` permission.

Reply:
just did the following test and only admin and monitor can access the stats
```
   @Test
   public void testStats() {
      restCreateAuthzCache("admin", "observer", "deployer", "application", "writer", "reader", "monitor");

      for (TestUser type : EnumSet.of(TestUser.ADMIN, TestUser.MONITOR)) {
         RestCacheClient cache = getServerTest().rest().withClientConfiguration(restBuilders.get(type)).get().cache(getServerTest().getMethodName());
         assertStatus(OK, cache.stats());
      }

      // Types with no access.
      for (TestUser type: EnumSet.complementOf(EnumSet.of(TestUser.ANONYMOUS, TestUser.ADMIN, TestUser.MONITOR))) {
         RestCacheClient cache = getServerTest().rest().withClientConfiguration(restBuilders.get(type)).get().cache(getServerTest().getMethodName());
         assertStatus(FORBIDDEN, cache.stats());
      }
   }
```

Reply:
@pruivo Thanks for the test, I am going to update the PR adding it too. IIUC, the `.getStats()` restricts the type without the `.permission()` in the resource endpoint because of:

https://github.com/infinispan/infinispan/blob/10220f7910942e2d2190c3933bd985ac0f6c493e/core/src/main/java/org/infinispan/security/impl/SecureCacheImpl.java#L892

Correct me if I am talking nonsense, but I think that for the distribution to work we need to either add a similar line somewhere or use the `.permission(MONITOR)`. What do you think?

Reply:
yeah, you are right. 
Since authz is enabled per cache, I would say the most correct solution is to check the permission at the cache level.

Reply:
I am retrieving the cache's auth manager and checking for permission if it is present.